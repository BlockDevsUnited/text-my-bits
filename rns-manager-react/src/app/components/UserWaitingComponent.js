import React from 'react';
import propTypes from 'prop-types';
import { Loader } from 'rimble-ui';

const UserWaitingComponent = ({ message, visible }) => {
  if (!visible) {
    return (<></>);
  }

  return (
    <div className="row waiting">
      <div className="col-md-6 offset-md-3">
        <Loader color="#008FF7" size="80px" className="loader-center" />
        <p>{message}</p>
      </div>
    </div>
  );
};

UserWaitingComponent.defaultProps = {
  message: '',
  visible: true,
};

UserWaitingComponent.propTypes = {
  message: propTypes.string,
  visible: propTypes.bool,
};

export default UserWaitingComponent;
