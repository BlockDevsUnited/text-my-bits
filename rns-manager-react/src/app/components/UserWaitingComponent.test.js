import React from 'react';
import { mount } from 'enzyme';

import UserWaitingComponent from './UserWaitingComponent';

describe('UserWaitingComponent', () => {
  it('renders and matches snapshot', () => {
    const component = mount(<UserWaitingComponent message="Please Wait!" />);
    expect(component.find('p').text()).toBe('Please Wait!');
    expect(component).toMatchSnapshot();
  });

  it('returns blank when visible is false', () => {
    const component = mount(<UserWaitingComponent message="Test Message!" visible={false} />);
    expect(component.html()).toEqual(null);
  });
});
