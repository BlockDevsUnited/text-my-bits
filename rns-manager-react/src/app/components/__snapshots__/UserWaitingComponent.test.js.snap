// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWaitingComponent renders and matches snapshot 1`] = `
<UserWaitingComponent
  message="Please Wait!"
  visible={true}
>
  <div
    className="row waiting"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <Loader
        bg="transparent"
        className="loader-center"
        color="#008FF7"
        size="80px"
      >
        <Styled(Box)
          bg="transparent"
          className="loader-center"
          color="#008FF7"
          size="80px"
        >
          <Box
            bg="transparent"
            className="sc-jXqrPq fhxevS loader-center"
            color="#008FF7"
            size="80px"
          >
            <styled.div
              bg="transparent"
              className="sc-jXqrPq fhxevS loader-center"
              color="#008FF7"
              size="80px"
              theme={
                Object {
                  "borderWidths": Array [
                    "0",
                    "1px",
                    "2px",
                    "4px",
                  ],
                  "borders": Array [
                    0,
                    "1px solid transparent",
                  ],
                  "breakpoints": Array [
                    "40em",
                    "52em",
                    "64em",
                  ],
                  "buttonSizes": Object {
                    "large": Object {
                      "fontSize": "1.5rem",
                      "height": "4rem",
                      "minWidth": "4rem",
                    },
                    "medium": Object {
                      "fontSize": "1rem",
                      "height": "3rem",
                      "minWidth": "3rem",
                    },
                    "small": Object {
                      "fontSize": "0.75rem",
                      "height": "2rem",
                      "minWidth": "2rem",
                      "padding": "0 1rem",
                    },
                  },
                  "buttons": Object {
                    "danger": Object {
                      "--contrast-color": "#fff",
                      "--main-color": "#DC2C10",
                    },
                    "primary": Object {
                      "--contrast-color": "#fff",
                      "--main-color": "#4E3FCE",
                      "backgroundColor": "#4E3FCE",
                      "color": "#fff",
                    },
                    "success": Object {
                      "--contrast-color": "#FFF",
                      "--main-color": "#28C081",
                    },
                  },
                  "colors": Object {
                    "background": "#fff",
                    "black": "#000",
                    "blacks": Array [
                      "rgba(0,0,0,.0125)",
                      "rgba(0,0,0,.025)",
                      "rgba(0,0,0,.05)",
                      "rgba(0,0,0,.1)",
                      "rgba(0,0,0,.2)",
                      "rgba(0,0,0,.3)",
                      "rgba(0,0,0,.4)",
                      "rgba(0,0,0,.5)",
                      "rgba(0,0,0,.6)",
                      "rgba(0,0,0,.7)",
                      "rgba(0,0,0,.8)",
                      "rgba(0,0,0,.9)",
                    ],
                    "blue": "#3259D6",
                    "danger": "#DC2C10",
                    "dark-gray": "#333",
                    "grey": "#CCC",
                    "info": "#36ADF1",
                    "light-gray": "#eee",
                    "light-silver": "#aaa",
                    "mid-gray": "#555",
                    "moon-gray": "#ccc",
                    "near-black": "#111",
                    "near-white": "#f4f4f4",
                    "primary": "#4E3FCE",
                    "primary-dark": "#3e32a4",
                    "primary-light": "#7165d7",
                    "silver": "#999",
                    "success": "#28C081",
                    "text": "#3F3D4B",
                    "transparent": "transparent",
                    "warning": "#FD9D28",
                    "white": "#fff",
                    "whites": Array [
                      "rgba(255,255,255,.0125)",
                      "rgba(255,255,255,.025)",
                      "rgba(255,255,255,.05)",
                      "rgba(255,255,255,.1)",
                      "rgba(255,255,255,.2)",
                      "rgba(255,255,255,.3)",
                      "rgba(255,255,255,.4)",
                      "rgba(255,255,255,.5)",
                      "rgba(255,255,255,.6)",
                      "rgba(255,255,255,.7)",
                      "rgba(255,255,255,.8)",
                      "rgba(255,255,255,.9)",
                    ],
                  },
                  "fontSizes": Array [
                    12,
                    14,
                    16,
                    20,
                    24,
                    32,
                    48,
                    64,
                  ],
                  "fontWeights": Array [
                    0,
                    300,
                    400,
                    600,
                    700,
                  ],
                  "fonts": Object {
                    "mono": "\\"Source Code Pro\\", -apple-system, monospace",
                    "sansSerif": "\\"Source Sans Pro\\", -apple-system, sans-serif",
                    "serif": "athelas, georgia, times, serif",
                  },
                  "letterSpacings": Array [
                    0,
                    1,
                    2,
                    4,
                    8,
                  ],
                  "lineHeights": Object {
                    "copy": 1.5,
                    "solid": 1,
                    "title": 1.25,
                  },
                  "messageStyle": Object {
                    "base": Object {
                      "backgroundColor": "#f4f4f4",
                      "borderColor": "#aaa",
                      "color": "#555",
                    },
                    "danger": Object {
                      "backgroundColor": "#fbe9e7",
                      "borderColor": "#DC2C10",
                      "color": "#841a09",
                    },
                    "info": Object {
                      "backgroundColor": "#eaf6fd",
                      "borderColor": "#36ADF1",
                      "color": "#206790",
                    },
                    "success": Object {
                      "backgroundColor": "#e9f8f2",
                      "borderColor": "#28C081",
                      "color": "#18734d",
                    },
                    "warning": Object {
                      "backgroundColor": "#fef5e9",
                      "borderColor": "#FD9D28",
                      "color": "#975e18",
                    },
                  },
                  "opacity": Object {
                    "disabled": 0.4,
                  },
                  "radii": Array [
                    "0",
                    "4px",
                    "8px",
                    "16px",
                    "32px",
                  ],
                  "shadows": Array [
                    "0",
                    "0px 2px 4px rgba(0, 0, 0, 0.1)",
                    "0px 8px 16px rgba(0, 0, 0, 0.1)",
                    "0 7px 14px rgba(50,50,93,.1)",
                  ],
                  "space": Array [
                    0,
                    4,
                    8,
                    16,
                    32,
                    64,
                    128,
                    256,
                  ],
                  "zIndices": Array [
                    0,
                    9,
                    99,
                    999,
                    9999,
                  ],
                }
              }
            >
              <div
                className="sc-AxirZ WKnMC sc-jXqrPq fhxevS loader-center"
                color="#008FF7"
                size="80px"
              >
                <svg
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
                    fill="currentColor"
                    fillRule="evenodd"
                    opacity="0.25"
                  />
                  <mask
                    height="24"
                    id="mask0"
                    mask-type="alpha"
                    maskUnits="userSpaceOnUse"
                    width="24"
                    x="0"
                    y="0"
                  >
                    <path
                      clipRule="evenodd"
                      d="M12 20.5714C16.7339 20.5714 20.5714 16.7339 20.5714 12C20.5714 7.26613 16.7339 3.42857 12 3.42857C7.26613 3.42857 3.42857 7.26613 3.42857 12C3.42857 16.7339 7.26613 20.5714 12 20.5714ZM12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
                      fill="currentColor"
                      fillRule="evenodd"
                    />
                  </mask>
                  <g
                    mask="url(#mask0)"
                  >
                    <path
                      d="M12 24V15.8824V12H24V24H12Z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </div>
            </styled.div>
          </Box>
        </Styled(Box)>
      </Loader>
      <p>
        Please Wait!
      </p>
    </div>
  </div>
</UserWaitingComponent>
`;
